### DISCUSSION TITLE
WatsonTcp.Settings.NoDelay purspose?

### DISCUSSION THREAD

**Discussion by HectorCarreno** [jchristn/WatsonTcp#293] [General]
I've been working with watsontcp recently, not sure what NoDelay works for exactly. I've just had some issues with cpu usage after watsontcp implementation. May somebody guide me with this please? 
Not sure if this delay helps to reduce possible cpu leakages at watson usage I couldn't find the exact description of this setting in the doc.

**Comment by jchristn** [comment]
The `NoDelay` setting controls use of Nagle's algorithm: https://en.wikipedia.org/wiki/Nagle%27s_algorithm and https://learn.microsoft.com/en-us/dotnet/api/system.net.sockets.tcpclient.nodelay?view=net-8.0

**Reply by HectorCarreno** [reply]
i've been reviewing Nagle's algorithm, now it make sense. 
I've tested it enabling <NoDelay> to skip the Nagle's algorithm but still having issues with watson and cpu high leakage (almost 100% sometimes). I'm profiling using jetbrains and trying to validate what's going on and what to do.
Thanks for your guidance jchristn.
